File: Ejercicios POO/Python/Herencia y Composicion/Cocheras.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Written By Ismael Heredia in the year 2016

from Autos import Auto

class Cochera(object):

   def __init__(self, ambientes="", dimensiones="",auto=""):
       self.__ambientes = ambientes
       self.__dimensiones = dimensiones
       self.__auto = auto
        
   def getAmbientes(self):
       return self.__ambientes

   def setAmbientes(self,ambientes = ""):
       self.__ambientes = ambientes

   def getDimensiones(self):
       return self.__dimensiones

   def setDimensiones(self, dimensiones = ""):
       self.__dimensiones = dimensiones
       
   def getAuto(self):
       return self.__auto

   def setAuto(self,auto = ""):
       self.__auto = auto
           
   def DatosCochera(self):
       contenido = ""
       contenido +=  "-- Datos de la cochera --\n\n[+] Ambientes : %d\n[+] Dimensiones : %s" % (self.__ambientes,self.__dimensiones)
       if self.__auto!=None:
           contenido += "\n-- Datos del auto --\n\n[+] Marca : %s\n[+] Numero de serie : %s" % (self.__auto.getMarca(),self.__auto.getNumero_serie())
       return contenido
          
   def destroy(self):
       self.__ambientes = None
       self.__dimensiones = None
       self.__auto = None

File: Ejercicios POO/PHP/Herencia y Composicion/Cocheras.php
<?php

// Written By Ismael Heredia in the year 2016

class Cochera
{
    
    private $ambientes;
    private $dimensiones;
	private $auto;
    
    public function __construct()
    {
        $this->ambientes   = "";
        $this->dimensiones = "";
		$this->auto = "";
    }
    
    public static function CreateCochera($ambientes,$dimensiones,$auto)
    {
        $instance                 = new self();
        $instance->ambientes   = $ambientes;
        $instance->dimensiones = $dimensiones;
		$instance->auto   = $auto;
        return $instance;
    }
    
    public function getAmbientes()
    {
        return $this->ambientes;
    }
    
    public function setAmbientes($ambientes)
    {
        $this->ambientes = $ambientes;
    }
    
    public function getDimensiones()
    {
        return $this->dimensiones;
    }
    
    public function setDimensiones($dimensiones)
    {
        $this->dimensiones = $dimensiones;
    }
	
    public function getAuto()
    {
        return $this->auto;
    }
    
    public function setAuto($auto)
    {
        $this->auto = $auto;
    }
        
    public function __destruct()
    {
        $this->ambientes   = "";
        $this->dimensiones = "";
		$this->auto = "";
    }
    
    public function DatosCochera()
    {
    	return "-- Datos de la cochera -- " . "\n\n"
                . "[+] Ambientes : " . $this->ambientes . "\n"
                . "[+] Dimensiones : " . $this->dimensiones . "\n" . $this->auto->getMarca();
    }
	
    public function DatosCocheraCompleto()
    {
    	$contenido =  "-- Datos de la cochera -- " . "\n\n"
                . "[+] Ambientes : " . $this->ambientes . "\n"
                . "[+] Dimensiones : " . $this->dimensiones . "\n";
				
		if(!is_null($this->auto)) {
        	$contenido = $contenido . "\n-- Datos del auto --" . "\n\n"
                . "[+] Marca : " . $this->auto->getMarca() . "\n"
                . "[+] Numero de serie : " . $this->auto->getNumero_serie() . "\n";	
		}
		return $contenido;		
    }
	
    
}

?>